projects/rocprofiler-sdk/source/include/rocprofiler-sdk/experimental/spm.h

```diff
@@ -0,0 +1,254 @@
+
+/**
+ * @brief SPM Profile Configurations
+ * @see rocprofiler_spm_create_counter_config for how to create.
+ */
+typedef struct ROCPROFILER_SDK_EXPERIMENTAL rocprofiler_spm_counter_config_id_t
+{
+    uint64_t handle;  ///< Opaque handle
+} rocprofiler_spm_counter_config_id_t;
+
+/**
+ * @brief (experimental) SPM parameter type and value.
+ *
+ **/
+typedef struct ROCPROFILER_SDK_EXPERIMENTAL rocprofiler_spm_parameter_t
+{
+    rocprofiler_spm_parameter_type_t type;   ///< SPM Parameter type
+    uint64_t                         value;  ///< SPM Parameter value
+} rocprofiler_spm_parameter_t;
+
+/**
+ * @brief (experimental) Create SPM Counter Configuration. A config is bound to an agent but can
+ *        be used across many contexts. The config has a fixed set of counters
+ *        that are collected (and specified by counter_list) and parameters. The available
+ *        counters for an agent can be queried using
+ *        ::rocprofiler_iterate_spm_supported_counters. An existing config
+ *        may be supplied via config_id to use as a base for the new config.
+ *        All counters and parameters in the existing config will be copied over to the new
+ *        config. The existing config will remain unmodified and usable with
+ *        the new config id being returned in config_id.
+ *
+ * @param [in] agent_id Agent identifier
+ * @param [in] counters_list List of GPU counters
+ * @param [in] counters_count Size of counters list
+ * @param [in] parameters Parameters list of rocprofiler_spm_parameter_type_t
+ * @param [in] parameter_count Number of parameters
+ * @param [in,out] config_id Identifier for GPU SPM counters group. If an existing
+                   config is supplied, that profiles counters and parameters will be copied
+                   over to a new config (returned via this id)
+ * @return ::rocprofiler_status_t
+ * @retval ROCPROFILER_STATUS_SUCCESS if config created
+ * @retval ROCPROFILER_STATUS_ERROR if config could not be created
+ * @retval ROCPROFILER_STATUS_ERROR_METRIC_NOT_VALID_FOR_AGENT if agent does not support an input
+ counter
+ * @retval ROCPROFILER_STATUS_ERROR_INCOMPATIBLE_ABI incompatible aqlprofile version is used
+ * @retval ROCPROFILER_STATUS_ERROR_NOT_IMPLEMENTED if the ROCPROFILER_SPM_BETA_ENABLED is not set
+ * @retval ROCPROFILER_STATUS_ERROR_EXCEEDS_HW_LIMIT if input counters exceed the hardware limit
+ * @retval ROCPROFILER_STATUS_ERROR_AGENT_NOT_FOUND if agent not found
+ * @retval ROCPROFILER_STATUS_ERROR_COUNTER_NOT_FOUND if an input counter is not found in metrics
+ file
+ */
+ROCPROFILER_SDK_EXPERIMENTAL
+rocprofiler_status_t
+rocprofiler_spm_create_counter_config(rocprofiler_agent_id_t               agent_id,
+                                      rocprofiler_counter_id_t*            counters_list,
+                                      size_t                               counters_count,
+                                      rocprofiler_spm_parameter_t*         parameters,
```

Comment:
```suggestion
                                      rocprofiler_spm_parameter_t**         parameters,
```
if the size of `rocprofiler_spm_parameter_t` can change, this needs to be an array of pointers